lib/pbench/agent/tool_meister.py

```diff
@@ -343,6 +221,181 @@ def wait(self):
+                self.args = None
+                self.env = None
+            else:
+                self.args = ["python2", f"{self.script_path}"]
+                self.env = os.environ.copy()
+                prev_path = os.environ.get("PYTHONPATH", "")
+                if prev_path:
+                    prev_path = f":{prev_path}"
+                self.env[
+                    "PYTHONPATH"
+                ] = "{self.install_path}/bindings:{self.install_path}/bindings/common{prev_path}"
+
+    def install(self):
+        if self.install_path is None:
+            return (1, "dcgm tool --inst argument missing")
+        elif self.args is None:
+            return (1, f"dcgm tool path, '{self.script_path}', not found")
+        return (0, "dcgm tool properly installed")
+
+    def start(self):
+        assert self.args is not None, "Logic bomb!  dcgm install had failed!"
+        self.logger.debug(self.args[1])
+        self.process = subprocess.Popen(
+            self.args,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.STDOUT,
+            env=self.env,
+            shell=True,
+        )
+        super().start()
+
+
+class NodeExporterTool(PersistentTool):
+    """NodeExporterTool - provide specifics for running the "node-exporter"
+    tool.
+
+    The only particular behavior is that we find the proper "node_exporter"
+    executable in our PATH.
+    """
+
+    def __init__(self, name, tool_opts, logger=None, **kwargs):
+        super().__init__(name, tool_opts, logger=logger, **kwargs)
+        executable = find_executable("node_exporter")
+        if executable is None:
+            self.args = None
+        else:
+            self.args = [executable]
+
+    def install(self):
+        if self.args is None:
+            return (1, "node_exporter tool not found")
+        return (0, "node_exporter tool properly installed")
+
+    def start(self):
+        assert self.args is not None, "Logic bomb!  node-exporter install had failed!"
+        self.logger.debug(self.args[0])
+        self.process = subprocess.Popen(
+            self.args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT
+        )
+        super().start()
+
+
+class PcpTool(PersistentTool):
+    """PcpTool - provide specifics for running the "pcp" tool, which is really the "pmcd" process.
+    """
+
+    # Default path to the "pmcd" executable.
+    _pmcd_path_def = "/usr/libexec/pcp/bin/pmcd"
+
+    def __init__(self, name, tool_opts, logger=None, **kwargs):
+        super().__init__(name, tool_opts, logger=logger, **kwargs)
+        pmcd_path = find_executable("pmcd")
+        if pmcd_path is None:
+            pmcd_path = self._pmcd_path_def
+        executable = os.access(pmcd_path, os.X_OK)
+        if executable:
+            self.args = [
+                pmcd_path,
+                "--foreground",
+                "--socket=./pmcd.socket",
+                f"--port={42042}",
```

Comment:
Does this work with the default pmlogger config?

Reply:
Actually, that is a bug.

Reply:
Fixed.